tests/unit/string_util_test.cpp

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/common/string_util.h>
+}
+
+TEST(strstr, basic)
+{
+   const char haystack[] = "haystack", needle[] = "needle", \
+   hay[] = "hay", empty[] = "";
+   ASSERT_EQ(strstr(empty, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, hay), haystack);
+}
+
+TEST(strncpy, basic)
+{
+   char dest[4], src[] = "a";
+   strncpy(dest, src,2);
```

Comment:
You’re calling libc’s strncpy() implementation. You should really rename this symbol to tilck_strncpy() and add an alias as you did for strstr().